flink-table/flink-table-runtime/src/main/java/org/apache/flink/table/runtime/functions/aggregate/JsonObjectAggFunction.java

```diff
@@ -128,6 +137,15 @@ public void merge(Accumulator acc, Iterable<Accumulator> others) throws Exceptio
                 assertKeyNotPresent(acc, key);
                 acc.map.put(key, other.map.get(key));
             }
+            for (final StringData key : other.retractMap.keys()) {
```

Comment:
IIUC, for retract streams, if -U comes before +U, +U might also be eliminated?

Reply:
Hi xuyang, thanks for your comments. It considers both the key and the corresponding value in the comparison. Therefore, I don't foresee any issues arising during the merge stage. However, the issue you mentioned could potentially occur at the local stage, such as within the retract function. I would fix it in retract function.

Reply:
Hi xuyang, Thanks for your comments. if +U is eliminated by -U, It means two messages has same key and value. It is an expected behaviour. However, you reminded me that the implementation here needs to be changed. Retraction is supposed to be done first in merge function.